pkg/kube/krt/collection.go

```diff
@@ -353,8 +353,9 @@ func (h *manyCollection[I, O]) onPrimaryInputEventLocked(items []Event[I]) {
 		i := a.Latest()
 		iKey := getTypedKey(i)
 
-		ctx := &collectionDependencyTracker[I, O]{h, nil, iKey}
-		results := slices.GroupUnique(h.transformation(ctx, i), getTypedKey[O])
+		ctx := &collectionDependencyTracker[I, O]{manyCollection: h, key: iKey}
+		res := h.transformation(ctx, i)
```

Comment:
Curiously, any reason the assignment was split up? I don't see `res` being used anywhere